### PR TITLE
webpack-multi-user-dev-server: generalize and open source

### PR DESCRIPTION
**Pull request by archive64**
Summary
---

Take our webpack dev server setup and do these things:
 * Generalize it to take a `usernameToConfigPath` parameter, rather than hardcoding our config paths.
 * Put in a separate, open source npm module.

CR notes
---

Compare this to `webpack.dev-server.js` in our main repo to see the changes.

cr_req 1

QA notes
---

Follow the "example" instructions in `README.md` and make sure it works.

```
git clone (this repo)

npm install
npm start

# in web browser
open localhost:8080/user1/bundle.js
open localhost:8080/user2/bundle.js
# should give you a js file that has a "console.log" at the bottom.
```

Closes https://github.com/iFixit/ifixit/issues/20128

### CODE REVIEW

File: example/user2/entry.js [hunk 1]
```diff
@@ -0,0 +1 @@
+console.log("I'm user2! I like inline sourcemaps.");
```
Comment (g1y) [inline example/user2/entry.js:1]: So what does user1 like then? I bet user1 likes long walks accross a fizzbuzz main loop, or programming languages that are palindromes.

**Comment by g1y** [comment]
CR :+1:

File: example/user2/entry.js (hunk 1 above)
Comment (archive64) [reply example/user2/entry.js:1]: Close! She actually likes long walks across fizzbuzz implemented with tail recursion.

**Comment by rachel-p** [comment]
QA :cocktail:

**Pull request merged**